OWL2/java/de/unibremen/informatik/OWL2Parser.java

```diff
@@ -402,32 +405,92 @@ void renderAsOmn(OWLOntology onto) {
 		}
 
 		void renderAsXml(OWLOntology onto) {
+			FileOutputStream fos = null;
+			FileInputStream fis = null;
+			OutputStreamWriter out = null;
+			InputStreamReader in = null;
+			char[] buf = new char[8192];
+			int c, n;
+			int chomp = -1;
 			try {
 				OWLXMLRenderer xmlren = new OWLXMLRenderer();
-				File tempFile = File.createTempFile("owlTemp_1", ".xml");
-				FileWriter buf = new FileWriter(tempFile);
+				String t = System.getenv("HETS_GZIP");
+				boolean gz = t == null || t.equalsIgnoreCase("on");
+				t = gz ? ".xml.gz" : ".xml";
```

Comment:
In general, yes.

However, `t` , `tmp`, `s`, `str` are commonly used as temporary string helper vars in a similar way as i, idx, k, n, m are used as integer helper vars within loops or to remember an index for a short time, or to make debugging easier, code more readable. _temporary_  in this context means "value for a very short lifetime in terms of LOC", i.e. set and used within a couple of lines in a way, that one is able to capture at once without scrolling. Therefore it isn't even worth to think about a very_long_descriptive_var_name for it and helps to avoid cluttering the code (if it would have a non-temp. var name, one would probably wonder, where else its value gets re-used, makes refactoring harder, ...). Last but not least using well-known temp var names may help the garbage collector a little bit, reduce memory fragmentation, etc. ...